.github/workflows/integration.yml

```diff
@@ -18,67 +20,71 @@ jobs:
           set -eux
           echo 'deb [trusted=yes] https://ftp.debian.org/debian/ testing main' | sudo tee /etc/apt/sources.list.d/testing.list
           sudo apt update
-          sudo apt install -y crun/testing podman/testing
+          sudo apt install -y crun/testing podman/testing just
 
       - uses: actions/checkout@v4
 
-      - name: Build bootc and bootc image
-        env:
-          TEST_OS: ${{ matrix.test_os }}
-        run: sudo -E TEST_OS=$TEST_OS tests/build.sh
+      - name: Set architecture variable
+        id: set_arch
+        run: echo "ARCH=$(arch)" >> $GITHUB_ENV
 
-      - name: Grant sudo user permission to archive files
+      - name: Build bootc and bootc image
         run: |
-          sudo chmod 0755 /tmp/tmp-bootc-build/id_rsa
+          sudo tests/build.sh ${{ matrix.test_os }}
+          # So the archiver works
+          sudo chmod 0755 target/id_rsa
 
       - name: Archive bootc disk image - disk.raw
-        if: matrix.test_runner == 'ubuntu-latest'
         uses: actions/upload-artifact@v4
         with:
-          name: PR-${{ github.event.number }}-${{ matrix.test_os }}-disk
-          path: /tmp/tmp-bootc-build/disk.raw
+          name: PR-${{ github.event.number }}-${{ matrix.test_os }}-${{ env.ARCH }}-disk
+          path: target/disk.raw
           retention-days: 1
 
       - name: Archive SSH private key - id_rsa
-        if: matrix.test_runner == 'ubuntu-latest'
         uses: actions/upload-artifact@v4
         with:
-          name: PR-${{ github.event.number }}-${{ matrix.test_os }}-id_rsa
-          path: /tmp/tmp-bootc-build/id_rsa
+          name: PR-${{ github.event.number }}-${{ matrix.test_os }}-${{ env.ARCH }}-id_rsa
+          path: target/id_rsa
           retention-days: 1
 
   test:
     needs: build
     strategy:
       matrix:
-        test_os: [fedora-41, fedora-42, fedora-43, centos-9]
+        #test_os: [fedora-42, fedora-43, centos-9, centos-10]
+        test_os: [centos-10]
         tmt_plan: [test-01-readonly, test-20-local-upgrade, test-21-logically-bound-switch, test-22-logically-bound-install, test-23-install-outside-container, test-24-local-upgrade-reboot]
 
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v4
 
+      - name: Set architecture variable
+        id: set_arch
+        run: echo "ARCH=$(arch)" >> $GITHUB_ENV
+
       - name: Install dependence
         run: |
           sudo apt-get update
-          sudo apt install -y qemu-kvm qemu-system
-          pip install --user tmt
+          sudo apt install -y qemu-kvm qemu-system just
+          sudo pip install --user tmt
```

Comment:
Can't use `sudo` in our case. The later `tmt` is running on rootless.

Reply:
OK back to running tmt as non-root.